tests/profile/ProfileUiTest.php

```diff
@@ -23,4 +22,62 @@ public function testProfileContainsUis()
         $this->assertEquals(1, $xpath->query('//userInterfaces/userInterface[@code="standard_representation_annotation_ui"]')->length, 'The user interface with code "standard_representation_annotation_ui" should exist.');
         $this->assertEquals(1, $xpath->query('//userInterfaces/userInterface[@code="movement_cataloguers_ui"]')->length, 'The user interface with code "movement_cataloguers_ui" should exist.');
     }
+
+    public function testValidRestrictionsForUserInterfacesAndScreens()
+    {
+        $uis_path = "/profile/userInterfaces/userInterface";
+        $uis = $this->xpath->query($uis_path);
+        $restrictions_count = 0;
+
+        /** @var DOMElement $ui */
+        foreach ($uis as $ui) {
+            $ui_table = $ui->getAttribute('type');
+            /** @var DOMElement $ui_or_screen_restriction */
+            foreach ($this->xpath->query("{$ui->getNodePath()}//restriction") as $ui_or_screen_restriction) {
+                $restrictions_count++;
+                $this->typeExistsForTable($ui_or_screen_restriction->getAttribute('type'), $ui_table, 'User interface or screen restriction: ' . $ui_or_screen_restriction->getNodePath());
+            }
+        }
+        $this->assertGreaterThan(0, $restrictions_count, 'At least one restriction should exist');
+    }
+
+    public function testValidRestrictionsForBundles()
+    {
+        $uis_path = "/profile/userInterfaces/userInterface";
+        $uis = $this->xpath->query($uis_path);
+        $restrictions_count = 0;
+
+        /** @var DOMElement $ui */
+        foreach ($uis as $ui) {
+            /** @var DOMElement $relation_restriction */
+            foreach ($this->xpath->query("{$ui->getNodePath()}/screens/screen/bundlePlacements/placement/settings/setting[@name='restrict_to_type']") as $relation_restriction) {
```

Comment:
You can combine these `foreach`:

    foreach ($this->xpath->query('/profile/userInterfaces/userInterface/screens/screen/bundlePlacements/placement/settings/setting[@name='restrict_to_type']') as $relation_restriction)...